packages/apps/src/Menu/ConnectButton.tsx

```diff
@@ -0,0 +1,140 @@
+// Copyright 2017-2022 @polkadot/apps authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { memo, useEffect, useState } from 'react';
+import styled from 'styled-components';
+
+import { extensionList, getUserInstalledExtensions, PreppedExtension } from '@polkadot/apps-config';
+import Icon from '@polkadot/react-components/Icon';
+
+import { useTranslation } from '../translate';
+import ConnectExtension from './ConnectExtension';
+
+const SHA_COL = 'rgba(34, 36, 38, 0.12)';
+const SHA_OFF = '5px';
+
+interface ConnectButtonProps {
+  className: string;
+}
+
+const ConnectButton = ({ className }: ConnectButtonProps) => {
```

Comment:
As per the comment on `function` vs `const`, this signature here should be -

```js
function ConnectButton ({ className }: Props): React.ReactElement<Props> {
...
```

... from here _really_ won't comment on these (as I suggested above already), however since it was the first component, may have been warranted here ...